File: 925.go
func isLongPressedName(name, typed string) bool {
	if len(name) > len(typed) {
		return false
	}

	var i int
	for _, b := range typed {
		b := byte(b)
		if i < len(name) && name[i] == b {
			i++
		} else if i == 0 && b != name[i] || i > 0 && b != name[i-1] {
			return false
		}
	}

	return i == len(name)
}

File: 925.rs
impl Solution {
    pub fn is_long_pressed_name(name: String, typed: String) -> bool {
        let nameBytes = name.as_bytes();

        let mut i = 0;
        let mut c = nameBytes[0];
        for &b in typed.as_bytes() {
            if i < nameBytes.len() && b == nameBytes[i] {
                i += 1;
            } else if b != nameBytes[i.saturating_sub(1)] {
                return false;
            }
        }

        i == nameBytes.len()
    }
}
